tests/Fist/regtest-spgr/TEST_FILES

```diff
@@ -0,0 +1,8 @@
+# runs are executed in the same order as in this file
+# the second field tells which test should be run in order to compare with the last available output
+# e.g. 0 means do not compare anything, running is enough
+#      1 compares with the space group name and number of operations provied in the SPGR section of the output file
+#      for details see cp2k/tools/do_regtest
+zro.inp                                      0      P2_1/c    4
```

Comment:
If you want to keep the comment (`P2_1/c    4`) then please move it to a separate line:
```
# P2_1/c    4
zro.inp                                      0
zro-filename.inp                             0      
```

Reply:
Is there a way to compare strings? Actually, I would like the regtest to check if the space group is properly identified after compilation?

Reply:
Unfortunately our `do_regtest` script can not readily compare strings.
However, the return value from [`spg_get_international`](https://spglib.github.io/spglib/api.html#spg-get-international) is actually the space group number. So, you could just output that one alongside the symbol.

Reply:
Silly me, I should have thought about that earlier.

I have an issue with spgr%international_symbolol, spgr%pointgroup_symbol, and spgr%schoenflies. There is a hidden character at the end that is visible with vim (^@). It also introduces a shift by one character visible with any editor. Is there a way to delete the last character like in python. I could do a TRIM(ADJUSTL(spgr%international_symbolol)) first, then how do I remove the last character?

Reply:
This is probably the null byte that terminates strings in C. In Fortran this is not needed because strings have fixed length. The conversion can be done like [this](https://github.com/cp2k/cp2k/blob/master/src/grid/grid_api.F#L1347-#L1352). I guess, we should add a helper function for this to [string_utilities.F](https://github.com/cp2k/cp2k/blob/master/src/common/string_utilities.F) some day.

Reply:
here you go: #1358

Reply:
@dev-zero @oschuett Thanks for the string converter. I did a fetch upstream and rebase to get it into my branch, but it seems it is not there. When I try to compile, I get an error that the function "strlcpy_c2f" is not not string_utilities module. I guess I did not use the correct git function.

On a different topic, the space group number is not accessible in Fortran from Spglib. There is not function for that. Ideally, accessing the dataset would solve this issue and spare three calls to Spglib space group identifying routine (not an issue for small systems but definitely a bottleneck for tens thousand+ systems). The module spglib_f08 contains an interface to access the dataset and even a fortran definition of the dataset, yet, during compilation I get an error that it does not exist. I cannot find what the issue is. The dataset type and get function are public:  https://github.com/spglib/spglib/blob/master/example/spglib_f08.f9

Reply:
@pcazade the link above was a PR, which I now just merged. If you fetch/rebase again now you should have the function. Usage of it can be seen in either `src/start/libcp2k.F` or `src/grid/grid_api.F`

Reply:
@dev-zero Thanks. Works like a charm.